devtools/x/src/tools.rs

```diff
@@ -12,17 +12,12 @@ pub struct Args {
 }
 
 pub fn run(args: Args, xctx: XContext) -> Result<()> {
-    let tools = xctx.config().tools();
-    let mut failed: bool = false;
-    for (key, value) in tools {
-        let success = match args.check {
-            false => installer::install_if_needed(key, value),
-            true => installer::check_installed(key, value),
-        };
-        failed = failed || !success;
+    let success = match args.check {
+        false => xctx.installer().install_all(),
+        true => xctx.installer().check_all(),
+    };
+    match success {
+        false => std::process::exit(1),
+        true => Ok(()),
```

Comment:
I would probably use if/else statements instead of matching on a bool but its not an issue if you'd prefer this